theano/tensor/extra_ops.py

```diff
@@ -647,9 +647,36 @@ def repeat(x, repeats, axis=None):
 
     .. versionadded:: 0.6
     """
-    return RepeatOp(axis=axis)(x, repeats)
+    if isinstance(repeats, (int, long, numpy.integer, list)):
+        repeats = numpy.asarray(repeats)
```

Comment:
You could probably call `as_tensor_variable` on it, and then check the number of dimensions of that instead.